### PR TITLE
refuse keeper sessions when client has seen higher zxid than server

### PR DESCRIPTION
**Pull request by mstetsyuk**
### Changelog category (leave one):
- Improvement

### Changelog entry (a [user-readable short description](https://github.com/ClickHouse/ClickHouse/blob/master/docs/changelog_entry_guidelines.md) of the changes that goes into CHANGELOG.md):
Add a keeper-server-side check during handshake to reject clients when `last_zxid_seen (provided by the client) > last_processed_zxid`. This prevents stale reads when clients reconnect to lagging replicas.

ZooKeeper does the same: https://github.com/apache/zookeeper/blob/fb43500ea53fa5b57b6dc549c6582ae0ac60d7bc/zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java#L1504

We should also:
1. Add an alert that checks the `ZooKeeperClientConnectionLoss` metric.
2. Add logic to the operator to block releases if the instance has at least one replica with `ZooKeeperClientConnectionLoss` equal 1.
3. Add logic to the auto-scaler to block horizontal and vertical scaling if the instance has at least one replica with `ZooKeeperClientConnectionLoss` equal 1.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

### Details

The client tracks the highest ZXID seen from responses and sends it during handshake. If the server's `last_processed_zxid` is lower, the connection is refused with `SESSION_REFUSED` error, forcing the client to try another server.

### CODE REVIEW

File: src/Interpreters/ServerAsynchronousMetrics.cpp [hunk 1]
```diff
@@ -367,6 +367,27 @@ void ServerAsynchronousMetrics::updateImpl(TimePoint update_time, TimePoint curr
         new_values["QueriesPeakMemoryUsage"] = { queries_peak_memory_usage, "Peak memory usage for queries, in bytes." };
     }
 
+    {
+        bool zookeeper_connection_loss = false;
+        try
+        {
+            const auto zookeeper = getContext()->getZooKeeper();
```
Comment (antonio2368) [inline src/Interpreters/ServerAsynchronousMetrics.cpp]: This will do 2 things I don't like:
- take ZooKeeper lock periodically
- slow down async metrics thread if it will wait the entire connection timeout period

Comment (mstetsyuk) [reply src/Interpreters/ServerAsynchronousMetrics.cpp]: good point, I've added a `ZooKeeperConnectionLossStartedTimestampSeconds` current metric and removed this logic from async metrics

**Review by antonio2368 — APPROVED** [review]
In general LGTM.
I'm interested if it can trigger some unwanted behavior.
Replica could be lagging to not have a certain ZXID so maybe it could cause unbalanced amount of connections between replicas. 🤔 

But let's see how it will behave.
